src/name-component.cpp

```diff
@@ -400,7 +400,13 @@ Component::compare(const Component& other) const
     return 0;
 
   // The components are equal length.  Just do a byte compare.
-  return std::memcmp(value(), other.value(), value_size());
+  int res = std::memcmp(value(), other.value(), value_size());
+
+  if (res < 0)
```

Comment:
Should we fix the tests and check negative values instead of just returning -1 for all negative values?